pinot-common/src/main/java/com/linkedin/pinot/common/partition/ReplicaGroupPartitionAssignment.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.partition;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+/**
+ * Class to represent a replica group partition assignment
+ *
+ */
+public class ReplicaGroupPartitionAssignment extends PartitionAssignment {
+  private static final String SEPARATOR = "_";
+
+  public ReplicaGroupPartitionAssignment(String tableName) {
+    super(tableName);
+  }
+
+  public ReplicaGroupPartitionAssignment(String tableName, Map<String, List<String>> partitionToInstances) {
+    super(tableName, partitionToInstances);
+  }
+
+  /**
+   * Compute total number of partitions in the mapping.
+   *
+   * @return the number of partitions
+   */
+  public int getNumPartitions() {
+    Set<String> partitions = new HashSet<>();
+    for (String key : getPartitionToInstances().keySet()) {
+      String partition = key.split(SEPARATOR)[0];
+      partitions.add(partition);
+    }
+    return partitions.size();
+  }
+
+  /**
+   * Compute total number of replica groups in the mapping.
+   *
+   * @return the number of replica groups
+   */
+  public int getNumReplicaGroups() {
+    Set<String> partitions = new HashSet<>();
```

Comment:
should this be Set<String> replicaGroups instead of partitions?

Reply:
fixed.